neqo-transport/src/streams.rs

```diff
@@ -24,13 +24,22 @@ use neqo_common::{qtrace, qwarn, Role};
 use std::{cell::RefCell, rc::Rc};
 use std::cmp::Ordering;
 
-struct StreamOrder {
-    sendorder: Option<i64>,
+#[derive(Copy, Clone)]
+pub struct StreamOrder {
+    pub sendorder: Option<i64>,
 }
 
-// We want highest to lowest
+// We want highest to lowest, with None being higher than any value
 impl Ord for StreamOrder {
     fn cmp(&self, other: &Self) -> Ordering {
```

Comment:
It should be possible to do this:

```rust
struct SendOrder(Option<i64>);

impl Ord for StreamOrder {
  fn cmp(&self, other: &Self) {
    if let (Some(a), Some(b)) = (self.0, other.0) {
      // We want reverse order (high to low) when both values are specified.
      other.0.cmp(self.0)
    } else {
      self.0.cmp(other.0)
    }
  }
}
```

see https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=c6ab1a1e00f3baa65e47f454be320d20

Reply:
With some tweaks